### PR TITLE
[regex] Fixing syntax based on user insight

### PR DESCRIPTION
**Pull request by emmbow**

### CODE REVIEW

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md?ref=9509dfd3ded49b23b408cd645147cc3a8c57b1fc)
- [ ] [enkidevs/curriculum/regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md?ref=9509dfd3ded49b23b408cd645147cc3a8c57b1fc)

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 1]
```diff
@@ -64,9 +64,9 @@ Which regex pattern would match the following string?
 
 ???
 
-* `/\w+/`
-* `/\W+/`
-* `/./`
+* `/\w/`
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:67]: `/\w+/` was the correct answer.

`/\w` will match `1`, `2`, `3` but it won't match `123`

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 2]
```diff
@@ -79,10 +79,10 @@ Which regex pattern would match the following string?
 
 ???
 
-* `/\W+/`
-* `/\w+/`
-* `/./`
-* `/12+/`
+* `/\W/`
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:82]: same here, `/\W/` will match `.`, `.`, `.` but it won't match `...`

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ We could use the following regex pattern to match these three strings:
 
 We could match just the numbers in the following strings by using this regex:
 
-`/\d+/`
+`/\d/`
 
 ```
 'abc**123**xyz' ✅
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:59]: so in this case, `/\d/` will match `1`, `2`, `3` individually, but it won't match `123`. 

we should say what we are matching (either `1`, `2`, `3` individually or `123` as a group)

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 4]
```diff
@@ -48,7 +48,7 @@ And would match any word character in a mixed string:
 
 > 💡 Using `\W` does the opposite of `\w` and would match all *non-word characters*.
 
-### `/\W*/`
+### `/\W/`
 
 '**!@£$%^&**' ✅
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:53]: do we want to match the entire string (`!@£$%^&`)? if so, we need to use `/\W*/` or `/\W+/`

if not, and we want to match each individual character in the string (`!`, `@`, ...) then we have to use `/\W/`

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md (hunk 1 above)
Comment (emmbow) [reply regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:67]: ah yep, the users comment made me worry about full matches so I took these `+` out. I will add them back in though!

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md (hunk 3 above)
Comment (emmbow) [reply regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:59]: Will add in some text to make this clearer

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 5]
```diff
@@ -18,56 +18,62 @@ aspects:
...
+\S matches all non-whitespace characters
+
+We could use the following regex pattern to match these three strings:
 
 ```
-' abc' ✅
-'	abc' ✅
-'    abc' ✅
-'abc' ❌
+/\s+abc/
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:42]:
```suggestion
/\s+abc/

' abc' ✅
'	abc' ✅
'    abc' ✅
'abc' ❌
```

let's move these back inside the code block :sweat_smile: 

using code ticks (or tables, not exactly sure which) removes any spaces

image

note: the other tables look good

File: regular-expressions/introduction/basic-syntax-i/regex-escaping-characters.md [hunk 6]
```diff
@@ -22,46 +22,48 @@ Because regex reserves the dot as a special character, we can't actually specify
...
+
+Similarly, if we wanted to use an actual backslash, we'd have to escape it too!
 
 ```
-'\backslash_demo' ✅
+/\\.+/
 ```
 
-Forward slashes must also always be escaped:
+Here, the /\\/ is escaping the backslash, and the /.+/ is saying that there are one or more characters following the backslash.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-escaping-characters.md:46]:
```suggestion
Here, the `/\\/` is escaping the backslash, and the `/.+/` is saying that there are one or more characters following the backslash.
```

File: regular-expressions/introduction/basic-syntax-i/regex-quantifiers-i.md [hunk 7]
```diff
@@ -20,30 +20,36 @@ aspects:
 
 Quantifiers allow us to build patterns that have additional characters without explicitly writing those characters.
 
-When we know how many times we want something repeated, we can put that number inside curly brackets `{}`. 
+When we know how many times we want something repeated, we can put that number inside curly brackets { }.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-quantifiers-i.md:23]:
```suggestion
When we know how many times we want something repeated, we can put that number inside curly brackets `{ }`. 
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 8]
```diff
@@ -26,34 +26,47 @@ Wildcards can be used to add flexibility when you don't want to specify characte
...
 
 > 💡 If we want to match only *word characters*, like letters, numbers and underscores, then we can use `\w`.
 
-### `/\w/`
+```
+/\w/
+```
 
-This pattern would match the following, as `\w` allows for capitalized letters:
+This pattern would match each character in the following text as \w allows for capitalized letters:
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:45]:
```suggestion
This pattern would match each character in the following text as `\w` allows for capitalized letters:
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 9]
```diff
@@ -26,34 +26,47 @@ Wildcards can be used to add flexibility when you don't want to specify characte
...
 
-'**abcABC123_**' ✅
+`a``b``c``A``B``C``1``2``3``_` ✅
 
-And would match any word character in a mixed string:
+If we want to match the entire string then we can use /\w+/ or /\w*/.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:49]:
```suggestion
If we want to match the entire string then we can use `/\w+/` or `/\w*/`.
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 10]
```diff
@@ -26,34 +26,47 @@ Wildcards can be used to add flexibility when you don't want to specify characte
...
 
-'**abc**!**f**@**g**£**jf**$**h**%^'
+`abcABC123_` ✅
 
-> 💡 Using `\W` does the opposite of `\w` and would match all *non-word characters*.
+\w can be used to match any word character in a mixed string:
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:53]:
```suggestion
`\w` can be used to match any word character in a mixed string:
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md [hunk 11]
```diff
@@ -26,34 +26,47 @@ Wildcards can be used to add flexibility when you don't want to specify characte
...
-'abc' ❌
 
-Note, the backslash is necessary syntax here. While some symbols are directly interpreted like special characters (e.g. `.`, `?` etc.), some only work like special characters if preceeded by a backslash. 
+> 💡 Using \W does the opposite of \w and would match all *non-word characters*.
+
+```
+/\W/
+```
+
+Similarly, We can use /\W+/ or /\W*/ to match an entire string:
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-i.md:64]:
```suggestion
Similarly, We can use `/\W+/` or `/\W*/` to match an entire string:
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 12]
```diff
@@ -18,56 +18,62 @@ aspects:
...
-Here are a few more wildcards that can be used when you want to be more specific than using a `.`, `\w` or `\W`!
+This would match the names that start with 'Fre', and then have zero or more word characters. 
 
-`\s` matches all whitespace characters - spaces, tabs and line-breaks.
+`Fred` ✅
+`Freya` ✅
+Francis ❌
 
-`\S` matches all non-whitespace characters
+Here are a few more wildcards that can be used when you want to be more specific than using a ., \w or \W!
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:33]:
```suggestion
Here are a few more wildcards that can be used when you want to be more specific than using a `.`, `\w` or `\W`!
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 13]
```diff
@@ -18,56 +18,62 @@ aspects:
...
 
-We could use the following regex pattern to match these three strings:
+\s matches all whitespace characters - spaces, tabs and line-breaks.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:35]:
```suggestion
`\s` matches all whitespace characters - spaces, tabs and line-breaks.
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 14]
```diff
@@ -18,56 +18,62 @@ aspects:
...
 
-`/\s+abc/`
+\S matches all non-whitespace characters
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:37]:
```suggestion
`\S` matches all non-whitespace characters
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 15]
```diff
@@ -18,56 +18,62 @@ aspects:
...
+/\s+abc/
 ```
 
-`\d` matches all digits
+| Text       | Match |
+|------------|-------|
+| ` abc`     | ✅     |
+| `   abc`   | ✅     |
+| `     abc` | ✅     |
+| abc        | ❌     |
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:50]: we'll move these in the code block so we don't need the table anymore
```suggestion
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 16]
```diff
@@ -18,56 +18,62 @@ aspects:
...
 
-`\D` matches all non-digits
+\d matches all digits
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:52]:
```suggestion
`\d` matches all digits
```

File: regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md [hunk 17]
```diff
@@ -18,56 +18,62 @@ aspects:
...
 
-We could match just the numbers in the following strings by using this regex:
+\D matches all non-digits
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-i/regex-wildcards-ii.md:54]:
```suggestion
`\D` matches all non-digits
```

File: regular-expressions/introduction/basic-syntax-ii/regex-anchors.md [hunk 18]
```diff
@@ -21,29 +21,33 @@ links:
...
+`success` ✅
+unsuccessful ❌
 
-`/.+\.$/`
+```
+/.+\.$/
+```
 
-The string can have one or more character but must end with a `'.'`.
+The string can have one or more character but must end with a '.'.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-anchors.md:45]:
```suggestion
The string can have one or more character but must end with a `'.'`.
```

looks a bit weird in the app without the back ticks

image

File: regular-expressions/introduction/basic-syntax-ii/regex-anchors.md [hunk 19]
```diff
@@ -83,12 +87,12 @@ Which regex pattern would match with the first line of text?
 ## Footnotes
 
 [1:Partial matching]
-Partial matches are when the regex pattern matches some, but not all of the text. An example of partial matching would be if we used `/\w/` to attempt to match all of the following text:
+Partial matches are when the regex pattern matches some, but not all of the text. An example of partial matching would be if we used /\w/ to attempt to match all of the following text:
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-anchors.md:90]:
```suggestion
Partial matches are when the regex pattern matches some, but not all of the text. An example of partial matching would be if we used `/\w/` to attempt to match all of the following text:
```

File: regular-expressions/introduction/basic-syntax-ii/regex-anchors.md [hunk 20]
```diff
@@ -83,12 +87,12 @@ Which regex pattern would match with the first line of text?
...
 
-"let's match this text..."
+'let's match this text...'
 
-`/\w/` partially matches on any word character, but it misses the punctuation marks and white spaces.
+/\w/ partially matches on any word character, but it misses the punctuation marks and white spaces.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-anchors.md:94]:
```suggestion
`/\w/` partially matches on any word character, but it misses the punctuation marks and white spaces.
```

File: regular-expressions/introduction/basic-syntax-ii/regex-anchors.md [hunk 21]
```diff
@@ -83,12 +87,12 @@ Which regex pattern would match with the first line of text?
...
 
-We could use the regex pattern `/.+/` to match the entire text.
+We could use the regex pattern /.+/ to match the entire text.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-anchors.md:96]:
```suggestion
We could use the regex pattern `/.+/` to match the entire text.
```

File: regular-expressions/introduction/basic-syntax-ii/regex-grouping.md [hunk 22]
```diff
@@ -18,29 +18,33 @@ aspects:
 ---
 ## Content
 
-We can group certain items using parenthesis `()`. 
+We can group certain items using parenthesis, ( ).
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-grouping.md:21]:
```suggestion
We can group certain items using parenthesis, `( )`. 
```

File: regular-expressions/introduction/basic-syntax-ii/regex-grouping.md [hunk 23]
```diff
@@ -18,29 +18,33 @@ aspects:
...
-'abc' ✅
-'abcabcabcabc' ✅
 ```
 
-We can build more complex groupings using the *or* operator, `|`.
+`abc` ✅
+`abcabcabcabc` ✅
 
-`/(a|b)c/`
+We can build more complex groupings using the *or* operator, '|'.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-grouping.md:41]:
```suggestion
We can build more complex groupings using the *or* operator, `|`.
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md [hunk 24]
```diff
@@ -18,44 +18,46 @@ aspects:
 ---
 ## Content
 
-The wildcards that we learnt about previously (`.`, `\w`, `\W`, `/s`, `/S`, `/d`, `/D`) are sometimes *too powerful.*
+The wildcards that we learnt about previously (., \w, \W, /s, /S, /d, /D) are sometimes *too powerful.*
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md:21]:
```suggestion
The wildcards that we learnt about previously (`.`, `\w`, `\W`, `/s`, `/S`, `/d`, `/D`) are sometimes *too powerful.*
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md [hunk 25]
```diff
@@ -18,44 +18,46 @@ aspects:
...
+/[cmf]an/
+```
 
-Our words would have to start with `'c'`, `'m'` *or* `'f'`, and end in `'an'`.
+Our words would have to start with `c`, `m` *or* `f`, and end in `an`.
 
 We can also use the `-` symbol to build an inclusive range:
 
-`/[0-9]/` means any digit between zero and nine, and is equivalent to `/\d/`.
+/[0-9]/ means any digit between zero and nine, and is equivalent to /\d/.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md:37]:
```suggestion
`/[0-9]/` means any digit between zero and nine, and is equivalent to `/\d/`.
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md [hunk 26]
```diff
@@ -18,44 +18,46 @@ aspects:
...
 
-`/[a-z]/` means any lowercase letter from 'a' to 'z'.
+/[a-z]/ means any lowercase letter from 'a' to 'z'.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md:39]:
```suggestion
`/[a-z]/` means any lowercase letter from 'a' to 'z'.
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md [hunk 27]
```diff
@@ -18,44 +18,46 @@ aspects:
...
 
-`/[a-zA-Z0-9_]/` is equivalent to `/\w/`
+/[a-zA-Z0-9_]/ is equivalent to /\w/
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-i.md:41]:
```suggestion
`/[a-zA-Z0-9_]/` is equivalent to `/\w/`
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-ii.md [hunk 28]
```diff
@@ -20,20 +20,20 @@ aspects:
...
-'def5678' ✅
 ```
 
+abc123 ❌
+`def5678` ✅
+
 This means that the following is true:
 
-`[^a-zA-Z0-9_]` is equivalent to `/W`
+/[^a-zA-Z0-9_]/ is equivalent to /W
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-ii.md:34]:
```suggestion
`/[^a-zA-Z0-9_]/` is equivalent to `/W`
```

File: regular-expressions/introduction/basic-syntax-ii/regex-ranges-ii.md [hunk 29]
```diff
@@ -20,20 +20,20 @@ aspects:
...
 
-`[^0-9]` is equivalent to `/D`
+/[^0-9]/ is equivalent to /D
```
Comment (kapnobatai137) [inline regular-expressions/introduction/basic-syntax-ii/regex-ranges-ii.md:36]:
```suggestion
`/[^0-9]/` is equivalent to `/D`
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md [hunk 30]
```diff
@@ -43,11 +43,11 @@ Let's try a stricter example:
 /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,6}/i
 ```
 
-`/[a-z0-9._%+-]+/` ensures that only email address-friendly characters are allowed.
+/[a-z0-9._%+-]+/ ensures that only email address-friendly characters are allowed.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md:46]:
```suggestion
`/[a-z0-9._%+-]+/` ensures that only email address-friendly characters are allowed.
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md [hunk 31]
```diff
@@ -43,11 +43,11 @@ Let's try a stricter example:
...
 
-`/@/` is literally just the '**@**' symbol.
+/@/ is literally just the '**@**' symbol.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md:48]:
```suggestion
`/@/` is literally just the '**@**' symbol.
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md [hunk 32]
```diff
@@ -43,11 +43,11 @@ Let's try a stricter example:
...
 
-`/[a-z0-9.-]+\.[a-z]{2,6}/` ensures that the domain also contains email address-friendly characters, and that there are between 2-6 characters in the domain extension. 
+/[a-z0-9.-]+\.[a-z]{2,6}/ ensures that the domain also contains email address-friendly characters, and that there are between 2-6 characters in the domain extension.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-i.md:50]:
```suggestion
`/[a-z0-9.-]+\.[a-z]{2,6}/` ensures that the domain also contains email address-friendly characters, and that there are between 2-6 characters in the domain extension. 
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md [hunk 33]
```diff
@@ -35,11 +35,11 @@ It looks like a lot, but don't be put off! 😅
 
 Let's take a look at each major chunk.
 
-`/[a-zA-Z0-9.!#$%&’*+/=?^_{|}~-]+/` gives all of the allowed email address characters, and ensures that there's at least one of them.
+/[a-zA-Z0-9.!#$%&’*+/=?^_{|}~-]+/ gives all of the allowed email address characters, and ensures that there's at least one of them.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md:38]:
```suggestion
`/[a-zA-Z0-9.!#$%&’*+/=?^_{|}~-]+/` gives all of the allowed email address characters, and ensures that there's at least one of them.
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md [hunk 34]
```diff
@@ -35,11 +35,11 @@ It looks like a lot, but don't be put off! 😅
...
 
-`/[a-zA-Z0-9-]+/` ensures that the host name is alphanumeric and has at least one character.
+/[a-zA-Z0-9-]+/ ensures that the host name is alphanumeric and has at least one character.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md:40]:
```suggestion
`/[a-zA-Z0-9-]+/` ensures that the host name is alphanumeric and has at least one character.
```

File: regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md [hunk 35]
```diff
@@ -35,11 +35,11 @@ It looks like a lot, but don't be put off! 😅
...
 
-`/(?:\.[a-zA-Z0-9-]+)*/` this is the optional part. It ensure that there must be zero or more instances of that group, so '**.com**' would match, but also '**.co.uk**'.
+/(?:\.[a-zA-Z0-9-]+)*/ this is the optional part. It ensure that there must be zero or more instances of that group, so `.com` would match, but also `.co.uk`.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-email-address-checker-ii.md:42]:
```suggestion
`/(?:\.[a-zA-Z0-9-]+)*/` this is the optional part. It ensure that there must be zero or more instances of that group, so `.com` would match, but also `.co.uk`.
```

File: regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md [hunk 36]
```diff
@@ -26,25 +26,25 @@ Password validation works best when we use a regex operator called a **positive
 
 This will search for the pattern inside the text to make sure it appears *at least once*, but the important thing is that the *order of where it appears doesn't matter*.
 
-Positive lookaheads start with `/(?=)/`. For example:
+Positive lookaheads start with /(?=)/. For example:
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md:29]:
```suggestion
Positive lookaheads start with `/(?=)/`. For example:
```

File: regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md [hunk 37]
```diff
@@ -26,25 +26,25 @@ Password validation works best when we use a regex operator called a **positive
...
 
-`/(?=.*[a-z])/` means that there must be at least one *lower case letter*.
+/(?=.*[a-z])/ means that there must be at least one *lower case letter*.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md:31]:
```suggestion
`/(?=.*[a-z])/` means that there must be at least one *lower case letter*.
```

File: regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md [hunk 38]
```diff
@@ -26,25 +26,25 @@ Password validation works best when we use a regex operator called a **positive
...
 
-`/(?=.*[A-Z])/` means that there must be at least one *upper case letter*.
+/(?=.*[A-Z])/ means that there must be at least one *upper case letter*.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md:33]:
```suggestion
`/(?=.*[A-Z])/` means that there must be at least one *upper case letter*.
```

File: regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md [hunk 39]
```diff
@@ -26,25 +26,25 @@ Password validation works best when we use a regex operator called a **positive
...
 
-`/(?=.*\d)/` means that there must be at least one *digit*.
+/(?=.*\d)/ means that there must be at least one *digit*.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md:35]:
```suggestion
`/(?=.*\d)/` means that there must be at least one *digit*.
```

File: regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md [hunk 40]
```diff
@@ -26,25 +26,25 @@ Password validation works best when we use a regex operator called a **positive
...
 
-`/(?=.*\W)/` means that there must be at least one *symbol*.
+/(?=.*\W)/ means that there must be at least one *symbol*.
```
Comment (kapnobatai137) [inline regular-expressions/introduction/practical-applications-i/regex-password-validation-using-positive-lookaheads.md:37]:
```suggestion
`/(?=.*\W)/` means that there must be at least one *symbol*.
```

**Pull request merged**